src/Storefront/Framework/Twig/TemplateDataExtension.php

```diff
@@ -93,7 +94,7 @@ public function getGlobals(): array
         ];
     }
 
-    protected function getThemeConfig(string $salesChannelId, ?string $themeId): array
+    protected function getThemeConfig(string $salesChannelId, ?string $themeId, ?string $themeName): array
```

Comment:
as this method is protected I think this is a break

Reply:
Yep, probably is. The commit history unfortunately does not give any indication on why it is protected and not private. Do you have any idea how to handle that or is it more of an FYI?

Reply:
probably to specific reason why this is protected :see_no_evil: 

maybe we can add a new private method which is used within the class. and this protected method calls the new private method to be backward compatible. and should of course be deprecated

Reply:
I see no problem in this change. Like @mitelg said, simply create a new function and mark this as deprecated.

The change of the ThemeCompiler::getThemePrefix is a problem.

Reply:
I have added the deprecation along with a new private method.
The static nature of `getThemePrefix` makes any checks on the parameters a bit weird, otherwise the theme could maybe be fetched with a matching on both the ID and the name.